src/sentry/static/sentry/app/utils/recreateRoute.tsx

```diff
@@ -1,18 +1,39 @@
+import {Location} from 'history';
+import {PlainRoute} from 'react-router';
 import {findLastIndex} from 'lodash';
 
 import replaceRouterParams from 'app/utils/replaceRouterParams';
 
-// Given a route object or a string and a list of routes + params from router, this will attempt to
-// recreate a location string while replacing url params.
-//
-// Can additionally specify the number of routes to move back
-//
-// See tests for examples
+type Options = {
+  routes: PlainRoute[];
+  location: Location;
+
+  // parameters to replace any route string parameters (e.g. if route is `:orgId`,
+  // params should have `{orgId: slug}`
+  params: {[key: string]: string};
+
+  /**
+   * The number of routes to to pop off of `routes
+   * Must be < 0
+   */
+  stepBack?: number;
```

Comment:
Is it worthwhile to use a union of reasonable values here? We could start with -1 to -5.

Reply:
Removed the runtime check as well